webapp/src/client.js

```diff
@@ -15,29 +16,45 @@ export default class Client {
         return this.doPost(`${this.url}/settings`, meeting);
     }
 
-    doGet = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
-        headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
-
-        const response = await request.
-            get(url).
-            set(headers).
-            accept('application/json');
-
-        return response.body;
+    doGet = async (url, headers = {}) => {
+        return this.doFetch(url, {headers});
     }
 
     doPost = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
-        headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
-
-        const response = await request.
-            post(url).
-            send(body).
-            set(headers).
-            type('application/json').
-            accept('application/json');
+        return this.doFetch(url, {
+            method : 'POST',
+            body: JSON.stringify(body),
+            headers: {
+                ...headers, 
+                'Content-Type' : 'application/json',
+            }
+        });
+    }
 
-        return response.body;
+    doFetch = async (url, { method = 'GET', body = null, headers = {} }) => {
+        const options = Client4.getOptions({
+            method,
+            body,
+            headers: {
+                ...headers,
+                'X-Timezoone-Offset' : new Date().getTimezoneOffset(),
```

Comment:
```suggestion
                'X-Timezone-Offset' : new Date().getTimezoneOffset(),
```

Reply:
This header is actually unnecessary for this plugin. It is usually used in other plugins to determine what time it is currently in the user's browser's timezone. It's used for plugin daily reminders specifically. The header is not checked on any API call in this plugin. I suggest removing the header here

Reply:
Removed the header. Thank you for noticing!